google/cloud/bigquery_storage_v1/writer.py

```diff
@@ -87,12 +88,20 @@ def __init__(
         self._closing = threading.Lock()
         self._closed = False
         self._close_callbacks = []
-        self._futures_queue = queue.Queue()
+        self._queue = queue.Queue()
         self._inital_request_template = initial_request_template
         self._metadata = metadata
+        self._resend = resend
 
-        # Only one call to `send()` should attempt to open the RPC.
-        self._opening = threading.Lock()
+
+        # if self._hibernating == True, this means the connection was closed by
+        # the server. The stream will try to reconnect if the message queue is 
+        # non-empty, or when the customer sends another request.
+        self._hibernating = False
+
+        # Only one call to `send()` should attempt to open the RPC. Use
```

Comment:
TODO: using reentrant lock may be against the original purpose that "Only one call to `send()` should attempt to open the RPC". Need to check:
1. Do we really need thread lock in _retry()?
2. If yes, we may need another send func without using any thread lock